dbt/adapters/athena/connections.py

```diff
@@ -57,6 +59,74 @@ def _collect_result_set(self, query_id: str) -> AthenaResultSet:
             retry_config=self._retry_config,
         )
 
+    def _clean_up_partitions(
+        self,
+        operation: str
+    ):
+        # Extract metadata from query
+        query_pattern = re.compile("delete\s+from\s+([^\s\.]+)\.([^\s\.]+)\s+where\s+([^;]+);")
+        query_search = query_pattern.search(operation)
+        if query_search is None:
+            raise OperationalError('Unable to extract metadata for cleaning up partitions from ' + operation)
+        database_name = query_search.group(1)
+        table_name = query_search.group(2)
+        where_condition = query_search.group(3)
+
+        # Look up Glue partitions & clean up
+        glue_client = boto3.client('glue')
+        s3_resource = boto3.resource('s3')
+        partitions = glue_client.get_partitions(
+            # CatalogId='awsdatacatalog',  # Using this caused permission error that 'glue:GetPartitions' is required
```

Comment:
@tuan-seek I am afraid this will need to be configurable

Reply:
@mrshu Thanks for checking out. Can you elaborate the scenarios that you would need to configure this CatalogID, instead of using the default `AWS Account ID`?

Reply:
@tuan-seek please take this with a grain of salt (the bigger the better).

It is my understanding that you can have multiple Glue Data Catalogs in one account. I don't think it happens often and by default the AWS Account ID is used -- that would probably explain why passing `awsdatacatalog` resulted in errors and why not passing it ended up working.

Here is what the [docs for Glue's `CreateDatabase` method](https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api-catalog-databases.html#aws-glue-api-catalog-databases-CreateDatabase) say:

> The ID of the Data Catalog in which to create the database. If none is provided, the AWS account ID is used by default.

All of the other Glue Database methods take the `CatalogId` as a parameter, so having a non-default one does seem like a real possibility.

I wouldn't worry too much about it for now though.